src/spider/storage/mysql/MySqlStorage.cpp

```diff
@@ -1874,7 +1927,16 @@ auto MySqlMetadataStorage::get_child_tasks(
         statement->setBytes(1, &id_bytes);
         std::unique_ptr<sql::ResultSet> res(statement->executeQuery());
         while (res->next()) {
-            children->emplace_back(fetch_full_task(static_cast<MySqlConnection&>(conn), res));
+            auto const fetch_task_result
+                    = fetch_full_task(static_cast<MySqlConnection&>(conn), res);
+            if (fetch_task_result.has_error()) {
+                static_cast<MySqlConnection&>(conn)->rollback();
+                return StorageErr{
+                        fetch_task_result.error(),
+                        fmt::format("Failed to fetch task {}", boost::uuids::to_string(id))
+                };
+            }
+            children->emplace_back(fetch_task_result.assume_value());
```

Comment:
```suggestion
            children->emplace_back(std::move(fetch_task_result.value()));
```
Same, use the move constructor and drop `const` in the result.